Compiler/src/inferencestate.jl

```diff
@@ -209,6 +209,11 @@ to enable flow-sensitive analysis.
 """
 const VarTable = Vector{VarState}
 
+struct StatementState
+    vtypes::Union{VarTable,Nothing}
+    bailed::Bool
```

Comment:
Naming nit: I think `had_latestworld` might be a more precise name than `bailed`?